lib/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -62,32 +64,62 @@ UsdSceneItem::Ptr UsdUndoRenameCommand::renamedItem() const
 	return fUfeDstItem;
 }
 
-bool UsdUndoRenameCommand::rename(SdfLayerHandle layer, const Ufe::Path& ufeSrcPath, const SdfPath& usdSrcPath, const SdfPath& usdDstPath)
+bool UsdUndoRenameCommand::renameRedo()
 {
-	InPathChange pc;
-	return internalRename(layer, ufeSrcPath, usdSrcPath, usdDstPath);
-}
+    // Copy the source path using CopySpec, and inactivate the source.
 
-bool UsdUndoRenameCommand::internalRename(SdfLayerHandle layer, const Ufe::Path& ufeSrcPath, const SdfPath& usdSrcPath, const SdfPath& usdDstPath)
-{
 	// We use the source layer as the destination.  An alternate workflow
 	// would be the edit target layer be the destination:
-	// layer = self._stage.GetEditTarget().GetLayer()
-	bool status = SdfCopySpec(layer, usdSrcPath, layer, usdDstPath);
+	// layer = self.fStage.GetEditTarget().GetLayer()
+	bool status = SdfCopySpec(fLayer, fUsdSrcPath, fLayer, fUsdDstPath);
 	if (status)
 	{
-		fStage->RemovePrim(usdSrcPath);
-		// The renamed scene item is a "sibling" of its original name.
-		fUfeDstItem = createSiblingSceneItem(
-			ufeSrcPath, usdDstPath.GetElementString());
-		// USD sends two ResyncedPaths() notifications, one for the CopySpec
-		// call, the other for the RemovePrim call (new name added, old name
-		// removed).  Unfortunately, the rename semantics are lost: there is
-		// no notion that the two notifications belong to the same atomic
-		// change.  Provide a single Rename notification ourselves here.
-		Ufe::ObjectRename notification(fUfeDstItem, ufeSrcPath);
-		Ufe::Scene::notifyObjectPathChange(notification);
+        auto srcPrim = fStage->GetPrimAtPath(fUsdSrcPath);
+        UFE_ASSERT_MSG(srcPrim, "Invalid prim cannot be inactivated.");
+        status = srcPrim.SetActive(false);
+
+        if (status) {
+            // The renamed scene item is a "sibling" of its original name.
+            auto ufeSrcPath = fUfeSrcItem->path();
+            fUfeDstItem = createSiblingSceneItem(
+                ufeSrcPath, fUsdDstPath.GetElementString());
+
+            Ufe::ObjectRename notification(fUfeDstItem, ufeSrcPath);
+            Ufe::Scene::notifyObjectPathChange(notification);
+        }
+    }
+    else {
+        UFE_LOG(std::string("Warning: SdfCopySpec(") +
+                fUsdSrcPath.GetString() + std::string(") failed."));
+    }
+
+	return status;
+}
+
+bool UsdUndoRenameCommand::renameUndo()
+{
+    // Regardless of where the edit target is currently set, switch to the
+    // layer where we copied the source prim into the destination, then
+    // restore the edit target.
+    auto editTarget = fStage->GetEditTarget();
+    fStage->SetEditTarget(fLayer);
+    bool status = fStage->RemovePrim(fUsdDstPath);
+    fStage->SetEditTarget(editTarget);
```

Comment:
Using a `UsdEditContext` might help simplify this a bit:
https://github.com/PixarAnimationStudios/USD/blob/be1a80f8cb91133ac75e1fc2a2e1832cd10d91c8/pxr/usd/usd/editContext.h#L40

Reply:
Excellent suggestion, and even better, exception-safe (if RemovePrim() throws).  Fixed in
https://github.com/Autodesk/maya-usd/pull/237/commits/f53f29c7258bc1d99897d843dbfef2e0316dd702